### PR TITLE
Address network duplication during import process

### PR DESCRIPTION
**Pull request by ibrokethecloud**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

**Problem:**

In case cases as reported by the linked issue, the nic's are duplicated in the imported VM.

There are different causes for this behaviour.

In case of openstack, if the network is dual stacked, then the api reports the network interface details twice, once with ipv4 and ipv6 addressing which causes the additional network mapping to appear. This is now fixed with a dedup of nics based on mac addresses

in case of vmware, this behaviour only happens when the same sourceNetwork appears in the import spec twice, without that multi-homed VM's get imported without any nic duplication. For example the following spec will trigger the duplication of nics. The easy fix is to dedup the source networks during the preflight check to allow the user to fix their network definitions and allow user to correct the spec.

```
apiVersion: migration.harvesterhci.io/v1beta1
kind: VirtualMachineImport
metadata:
  name: ubuntu-import
  namespace: default
spec:
  virtualMachineName: "gm-ubuntu-test"
  networkMapping:
  - sourceNetwork: "VM Network"
    destinationNetwork: "default/workload"
  - sourceNetwork: "VM Network"
    destinationNetwork: "default/workload"
  sourceCluster:
    name: wrangler
    namespace: default
    kind: VmwareSource
    apiVersion: migration.harvesterhci.io/v1beta1
```

**Solution:**

PR introduces two minor fixes:
* dedup network info for openstack vm's based on mac addresses to avoid creating a nic in imported VM for each match of source to destination network
* dedup source network for VMs during preflight checks to avoid same nic being mapped twice for the same source network

**Related Issue:**
https://github.com/harvester/harvester/issues/6451
**Test plan:**

### CODE REVIEW

File: pkg/source/openstack/client.go [hunk 1]
```diff
@@ -54,6 +54,15 @@ type ExtendedVolume struct {
 	VolumeImageMetadata map[string]string `json:"volume_image_metadata,omitempty"`
 }
 
+// ExtendedServer The original `Server` structure does not contain the `Description` field.
```
Comment (votdev) [inline pkg/source/openstack/client.go:57]: I think this is wrong. This and other code in your PR is from https://github.com/harvester/vm-import-controller/pull/40.

Comment (ibrokethecloud) [reply pkg/source/openstack/client.go:57]: 🤦 let me fix it

**Review by votdev — APPROVED** [review]
LGTM

**Review by starbops — APPROVED** [review]
LGTM, thank you.

**Comment by ibrokethecloud** [comment]
@mergifyio backport v1.3

**Pull request merged**